gammapy/astro/darkmatter/utils.py

```diff
@@ -52,3 +66,181 @@ def compute_jfactor(self):
+        jfact=1,
+        absorption_model=None,
+        z=0,
+        k=2,
+        xsection=None,
+    ):
+
+        self.dataset = dataset
+        self.masses = masses
+        self.channels = channels
+        self.jfact = jfact
+        self.absorption_model = absorption_model
+        self.z = z
+        self.k = k
+
+        if not xsection:
+            xsection = DMAnnihilation.THERMAL_RELIC_CROSS_SECTION
+        self.xsection = xsection
+
+        self._spatial_model = dataset.model.spatial_model
+        self._geom = dataset.counts.geom
+        self._exposure = dataset.exposure
+        self._background_model = dataset.background_model
+        self._psf = dataset.psf
+        self._edisp = dataset.edisp
+
+        self._counts_map = WcsNDMap(self._geom, np.random.poisson(dataset.npred().data))
+
+    def run(self, optimize_opts=None, covariance_opts=None):
+        """Run the SigmaVEstimator for all channels and masses.
+
+        Parameters
+        ----------
+        optimize_opts : dict
+            Options passed to `Fit.optimize`.
+        covariance_opts : dict
+            Options passed to `Fit.covariance`.
+
+        Returns
+        -------
+        result : dict
+            Dict with results as `~astropy.table.Table` objects for each channel.
+        """
+
+        result = {}
+        sigma_unit = ""
+        for ch in self.channels:
+            tablerows = []
+            log.info("Channel: {}".format(ch))
+            for mass in self.masses:
+                row = {}
+                log.info("Mass: {}".format(mass))
+                DMAnnihilation.THERMAL_RELIC_CROSS_SECTION = self.xsection
+                spectral_model = DMAnnihilation(
+                    mass=mass,
+                    channel=ch,
+                    scale=1,
+                    jfactor=self.jfact,
+                    z=self.z,
+                    k=self.k,
+                )
+                if self.absorption_model:
+                    spectral_model = AbsorbedSpectralModel(
+                        spectral_model, self.absorption_model, self.z
+                    )
+                spatial_model = self._spatial_model
+                flux_model = SkyModel(
+                    spatial_model=spatial_model, spectral_model=spectral_model
+                )
+                if isinstance(self._spatial_model, SkyPointSource):
+                    flux_model.parameters["lat_0"].frozen = True
+                    flux_model.parameters["lon_0"].frozen = True
+
+                dataset_loop = MapDataset(
+                    model=flux_model,
+                    counts=self._counts_map,
+                    exposure=self._exposure,
+                    background_model=self._background_model,
+                    psf=self._psf,
+                    edisp=self._edisp,
+                )
+                try:
+                    fit = Fit(dataset_loop)
+                    fit.datasets.parameters.apply_autoscale = False
```

Comment:
As you use the `fit.likelihood_profile()` method now, it's not necessary to switch off the autoscaling. Maybe just remove...